### ISSUE TITLE
Brush rect is flat 

### ISSUE THREAD

**Issue by gordonwoodhull** [dc-js/dc.js#1134]
@paulbriton reported on gitter:
> I'm having trouble with the brush and the include of dc.js with RequireJs.. When I include the libs with the classic `<script>` every thing is ok but when it's done with requirejs brush background isn't showed. After further investigations it seems that the height property is not set on the <rect> elements when using requirejs. There is no js errors, and when I set manually the height of both element it's ok. Any explantions ?

![Bug](https://files.gitter.im/dc-js/dc.js/Uwoa/bug.png)

Bug is present since beta 16.

**Comment by paulbriton** [comment]
Here is two commits between beta 15 and 16 about the brush :

- [https://github.com/dc-js/dc.js/commit/354b89010eae538877fd78aeefebd2865a061d2f](url)
- [https://github.com/dc-js/dc.js/commit/2e4052f5752ef43e2a36837ca89c8f361cdc6f9b](url)

I will try later this week to give a fiddle demonstrating the problem.

By the way, thank's for your availability.

**Comment by paulbriton** [comment]
I've tried to reproduce it in a fiddle but I didn't manage to have the same error. 

I know that without a reproductible case, it's hard to correct, but the error is still present in the latest releases.

**Comment by gordonwoodhull** [comment]
The commits between those releases are mostly to do with resizing transitions. Not that it's a solution, but does the brush show up if you set `transitionDuration(0)`? 

Other than that, if you send me a link to my private email (on my github profile) I can debug it confidentially. It takes a lot of time just guessing but it should be easy to track down in the debugger.

**Comment by paulbriton** [comment]
The `transitionDuration(0)` doesn't show the brush.

I know that giving you the access should be easier but it's not my call since we have confidential information on this site. I will ask and let you know.

As always, thank's for your availability.

**Comment by gordonwoodhull** [comment]
@paulbriton tracked it down to this line:

```js
        gBrush.selectAll('.brush rect')
            .attr('height', brushHeight());
```
https://github.com/dc-js/dc.js/blob/a27ce29a24b8cfa7aa85168388a91ada6dc17203/src/coordinate-grid-mixin.js#L1012

which used to `selectAll('rect')` but was changed in 354b89010e when the intent was just to change the brush handle size. Looks like an unintentional change.

What's curious is that when I try this in the debugger, it still selects all four child rects, even though [selection.selectAll is documented as selecting only descendants](https://github.com/d3/d3-3.x-api-reference/blob/master/Selections.md#selectAll). 

So I think the code is incorrect but may work with some versions of d3 but not others (?) I can't explain otherwise how the use of require would affect this code.

**Comment by gordonwoodhull** [comment]
Thanks @paulbriton, released this in 2.0 beta 31
